scripts/watch_frontend_lib.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# This script watches for files changes and files created with the .ts, .tsx, .js, and .jsx
+# extensions. If any of those events occurs, this will automatically rebuild the frontend streamlit library.
+# This script is useful if you're mainly changing code within frontend/lib/src and you don't want to constantly
+# run yarn buildLib and this will automatically "hot reload" your code by rebuilding the library.
+
+import os
+import subprocess
+
+from watchdog.events import FileSystemEventHandler
+from watchdog.observers import Observer
+
+# Directory to monitor
+DIRECTORY = os.path.dirname(__file__).join(["./frontend/lib"])
```

Comment:
Python's `str.join` function is intended for joining lists of strings into one single string. This is a pretty weird usage of it: we're just concatenating two strings, and it would be clearer to use string interpolation, or os.path.join, or the Path module.

```python
DIRECTORY = os.path.join(os.path.dirname(__file__), "./frontend/lib")
```

Reply:
So I tried testing this out and it didn't work as the directory that it started checking was: 
`scripts/./frontend/lib`

However, I think what we need is:
`scripts/../frontend/lib` so the new DIRECTORY line would look like this if this is fine:
DIRECTORY = os.path.join(os.path.dirname(__file__), "../frontend/lib")

Can you confirm if this is ok?

Reply:
Yep, that's fine!

Alternately, you could use the Path module and do something like `Path(__file__).parent.parent / "frontend/lib"`